.github/workflows/build_wheels.yml

```diff
@@ -38,9 +38,9 @@ jobs:
       run: |
         python -m cibuildwheel --output-dir wheelhouse
 
-    - uses: actions/upload-artifact@v3
+    - uses: actions/upload-artifact@v4
       with:
-        name: wheels
+        name: wheels-${{ strategy.job-index }}
```

Comment:
maybe matrix.os instead of job-index is easier to parse?

Also could you please commit with message 'build wheels' to trigger the build and check that they run properly?

Reply:
Ah, yes I remember why this pattern is needed now. If you just do `matrix.os` this isn't sufficiently unique in [`.github/workflows/build_wheels.yml`](https://github.com/PythonOT/POT/blob/38922c0bd72df3c332d65b5c40b6f6a14c4fc53e/.github/workflows/build_wheels.yml) as both

https://github.com/PythonOT/POT/blob/38922c0bd72df3c332d65b5c40b6f6a14c4fc53e/.github/workflows/build_wheels.yml#L11-L18

and 

https://github.com/PythonOT/POT/blob/38922c0bd72df3c332d65b5c40b6f6a14c4fc53e/.github/workflows/build_wheels.yml#L47-L53

will have the same name and so will result in errors of

```
Error: Failed to CreateArtifact: Received non-retryable error: Failed request: (409) Conflict: an artifact with this name already exists on the workflow run
```

So you need something more unique. In [`boost-histogram` we use `wheels-${{ strategy.job-index }}`](https://github.com/scikit-hep/boost-histogram/blob/69a9399e0d97c7d4e95678715b59232a0f4ee2a3/.github/workflows/wheels.yml#L136) and it works well.

Reply:
I see. on the other hand both build wheels and build all wheels should not run simultaneously because of the condition. I don't know how you managed that ;)

Also the only one that run on master after a merge is in build_wheels_weekly and this one do not have this kinf of problems. Let us use job-index in build_wheels to avoid the problem and matrix.ox in the weeky file (that also run for all merge on master) so that we have easier to handle filenames when releasing maybe?